src/main/java/Storage/TaskList.java

```diff
@@ -43,9 +43,26 @@ public boolean isEmpty() {
         return this.db.isEmpty();
     }
 
-    public void showTasks() {
+    public String getTasks() {
+        String str = "";
         for (int i = 1; i <= this.db.size(); i++) {
-            System.out.println(i + "." + this.db.get(i-1));
+            str += (i + "." + this.db.get(i - 1) + "\n");
         }
+        return str.substring(0, str.length() - 1);
+    }
+
+    /**
+     * Returns all the matching tasks in string representation.
+     * @param description The description to be matched.
+     * @return The string representation of all matching tasks.
+     */
+    public String getMatchingTasks(String description) {
+        String str = "";
+        for (int i = 1; i <= this.db.size(); i++) {
+            if (this.db.get(i - 1).getDescription().contains(description)) {
+                str += (i + "." + this.db.get(i - 1) + "\n");
```

Comment:
Could this line possibly be abstracted out as a function?

Reply:
Thank you for the suggestion! I will take note of this in my future commits!